res/controllers/midi-components-0.1.js

```diff
@@ -0,0 +1,1136 @@
+/**
+ * Components JS library for Mixxx
+ * Documentation is on the Mixxx wiki at
+ * http://mixxx.org/wiki/doku.php/components_js
+ *
+ * Copyright (C) 2017 Be <be.0@gmx.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *
+ *
+ * This library depends on Lodash, which is copyright JS Foundation
+ * and other contributors and licensed under the MIT license. Refer to
+ * the lodash.mixxx.js file in this directory for details.
+**/
+
+(function(global) {
+    var Component = class {
+        constructor(options) {
```

Comment:
This is still not supported.

Reply:
Ah, damn, I forgot. Do you remember where that was discussed?

Reply:
Are you sure? `QJSEngine::importModule` was added in Qt 5.12: https://doc.qt.io/qt-5/qjsengine.html#importModule

Reply:
The problem is that even though we're using QJSEngine, using `importModule` results in a breaking API change because we can only access exported symbols from the module, but since our mappings don't export anything, we can't access the controller.
Besides, we already have ControllerScriptModuleEngine but its not used anywhere because its part the not-yet-completed controller scripting engine rewrite.

Reply:
Unfortunately the `import` JS keyword only works when code is executed via `QJSEngine::importModule`, but ControllerScriptEngineLegacy uses `QJSEngine::evaluate`.